### PR TITLE
[ONNX] model_type flag is only supported under `SKIP_XFAIL_SUBTESTS`

### LINKED ISSUE #120918: [ONNX] `test_fx_op_consistency.py` some cases success without running
Found in ops: nn.functional.instance_norm from https://github.com/pytorch/pytorch/pull/120866

### PR DESCRIPTION
**Pull request by titaiwangms**
Fixes #120918 

To address the confusion that developers usually have on which list to put xfail and skip. This PR provides guidance that `model_type` and `matcher` specified xfail/skip should go to `SKIP_XFAIL_SUBTESTS`.

### CODE REVIEW

**Review by BowenBao — APPROVED** [review]

File: test/onnx/test_fx_op_consistency.py [hunk 1]
```diff
@@ -133,6 +133,8 @@ def skip_torchlib_forward_compatibility(
 #     2a. If a test is now failing because of xpass, because some previous errors
 #     are now fixed, removed the corresponding xfail.
 #     2b. If a test is not failing consistently, use skip.
+# NOTE: EXPECTED_SKIPS_OR_FAILS only supports dtypes. If a matcher or model_type
```
Comment (BowenBao) [inline test/onnx/test_fx_op_consistency.py:136]: might be better if code can enforce this.

File: test/onnx/test_fx_op_consistency.py [hunk 2]
```diff
@@ -133,6 +133,8 @@ def skip_torchlib_forward_compatibility(
...
+# is needed, use the SKIP_XFAIL_SUBTESTS list further down below.
 EXPECTED_SKIPS_OR_FAILS: Tuple[onnx_test_common.DecorateMeta, ...] = (
```
Comment (thiagocrepaldi) [inline test/onnx/test_fx_op_consistency.py:138]:
```suggestion
EXPECTED_SKIPS_OR_FAILS_FOR_DTYPES: Tuple[onnx_test_common.DecorateMeta, ...] = (
```

File: test/onnx/test_fx_op_consistency.py [hunk 3]
```diff
@@ -1351,6 +1346,8 @@ def skip_torchlib_forward_compatibility(
 )
 # fmt: on
 
+# NOTE: The xfail and skip with a matcher function or model_type should be
+# at under the `SKIP_XFAIL_SUBTESTS` section.
 SKIP_XFAIL_SUBTESTS: tuple[onnx_test_common.DecorateMeta, ...] = (
```
Comment (thiagocrepaldi) [inline test/onnx/test_fx_op_consistency.py:1351]:
```suggestion
SKIP_XFAIL_SUBTESTS_FOR_MODEL_TYPE: tuple[onnx_test_common.DecorateMeta, ...] = (
```

**Review by thiagocrepaldi — APPROVED** [review]
+1 to Bowen's comment, but maybe renaming the tuples to make that clear there too is an incremental improvement

**Pull request closed without merging**